libcalico-go/lib/validator/v3/validator.go

```diff
@@ -1327,6 +1329,15 @@ func validateRule(structLevel validator.StructLevel) {
 			} else {
 				v4 = v4 || cidr.Version() == 4
 				v6 = v6 || cidr.Version() == 6
+				
+				// Check for universal CIDR in negated context, which creates logical contradictions
+				if isNegatedField {
+					if (cidr.Version() == 4 && n == "0.0.0.0/0") ||
+					   (cidr.Version() == 6 && n == "::/0") {
```

Comment:
CI is not happy with this indentation

```suggestion
						   (cidr.Version() == 6 && n == "::/0") {
```